core/server/models/user.js

```diff
@@ -75,21 +75,45 @@ User = ghostBookshelf.Model.extend({
         });
     },
 
+    /**
+     * Lookup Gravatar if email changes to update image url
+     * Generating a slug requires a db call to look for conflicting slugs
+     */
     saving: function saving(newPage, attr, options) {
-        /*jshint unused:false*/
-
-        var self = this;
+        var self = this,
+            ops = [];
 
         ghostBookshelf.Model.prototype.saving.apply(this, arguments);
 
-        if (this.hasChanged('slug') || !this.get('slug')) {
-            // Generating a slug requires a db call to look for conflicting slugs
-            return ghostBookshelf.Model.generateSlug(User, this.get('slug') || this.get('name'),
-                {status: 'all', transacting: options.transacting, shortSlug: !this.get('slug')})
-                .then(function then(slug) {
-                    self.set({slug: slug});
+        if (self.hasChanged('email')) {
+            ops.push(function lookUpGravatar() {
+                return gravatar.lookup({
+                    email: self.get('email')
+                }).then(function (response) {
+                    if (response && response.image) {
+                        self.set('image', response.image);
+                    }
                 });
+            });
+        }
+
+        if (this.hasChanged('slug') || !this.get('slug')) {
+            ops.push(function generateSlug() {
+                return ghostBookshelf.Model.generateSlug(
+                    User,
+                    self.get('slug') || self.get('name'),
+                    {
+                        status: 'all',
+                        transacting: options.transacting,
+                        shortSlug: !self.get('slug')
+                    })
+                    .then(function then(slug) {
+                        self.set({slug: slug});
+                    });
+            });
         }
+
+        return utils.sequence(ops);
```

Comment:
ok good point. i wasn't thinking about running them in parallel, but actually that is possible. So will change 👍